web/templates/stream_settings/stream_types.hbs

```diff
@@ -1,76 +1,107 @@
-            <div class="settings-radio-input-parent">
-                <label class="radio">
-                    <input type="radio" name="privacy" value="{{ this.code }}" {{#if (eq this.code ../stream_privacy_policy) }}checked{{/if}} />
-                    <b>{{ this.name }}:</b> {{ this.description }}
-                </label>
-            </div>
-        {{/each}}
+    <div class="input-group stream-privacy-values">
+        <div class="alert stream-privacy-status"></div>
+        <label>{{t 'Who can access the channel?'}}
+            {{> ../help_link_widget link="/help/channel-permissions" }}
+        </label>
+
+        <div class="stream-privacy_choices prop-element" id="id_stream_privacy" data-setting-widget-type="radio-group" data-setting-choice-type="string">
+            {{#each stream_privacy_policy_values}}
+                <div class="settings-radio-input-parent">
+                    <label class="radio">
+                        <input type="radio" name="privacy" value="{{ this.code }}" {{#if (eq this.code ../stream_privacy_policy) }}checked{{/if}} />
+                        <b>{{ this.name }}:</b> {{ this.description }}
+                    </label>
+                </div>
+            {{/each}}
+        </div>
     </div>
-</div>
 
-<div class="default-stream">
-    {{> ../settings/settings_checkbox
-      prefix="id_"
-      setting_name="is_default_stream"
-      is_checked=check_default_stream
-      label=(t "Default channel for new users")
-      help_link="/help/set-default-channels-for-new-users"
-      }}
-</div>
+    {{#if ask_to_announce_stream}}
+        <div id="announce-new-stream">
+            {{>announce_stream_checkbox }}
+        </div>
+    {{/if}}
 
-<div class="input-group">
-    <label class="settings-field-label">{{t 'Who can post to the channel?'}}
-        {{> ../help_link_widget link="/help/channel-posting-policy" }}
-    </label>
-    <select name="stream-post-policy" class="stream_post_policy_setting prop-element settings_select bootstrap-focus-style" id="id_stream_post_policy" data-setting-widget-type="number">
-        {{#each stream_post_policy_values}}
-            <option value="{{this.code}}" {{#if (eq this.code ../stream_post_policy) }}selected{{/if}}>
-                {{ this.description}}
-            </option>
-        {{/each}}
-    </select>
+    <div class="default-stream">
+        {{> ../settings/settings_checkbox
+          prefix="id_"
+          setting_name="is_default_stream"
+          is_checked=check_default_stream
+          label=(t "Default channel for new users")
+          help_link="/help/set-default-channels-for-new-users"
+          }}
+    </div>
 </div>
 
-{{> ../dropdown_widget_with_label
-  widget_name=can_remove_subscribers_setting_widget_name
-  label=(t 'Who can unsubscribe others from this channel?')
-  value_type="number"}}
+<div class="advanced-configurations-container {{#if is_stream_edit}}settings-subsection-parent settings-sujal{{/if}}">
```

Comment:
Can we just keep these classes hare and below irrespective of whether it is stream creation or stream edit form? Also, there is a typo here I guess with `settings-sujal` class.

Reply:
I forgot to remove `settings-sujal` 😅sorry for that. If we remove `{{#if is_stream_edit}}` then at the time of stream creation it will give this error :

>[Also, when creating a stream, it is giving an error due to this change.](https://github.com/zulip/zulip/pull/29433#:~:text=and%20discard%22%20buttons.-,Also%2C%20when%20creating%20a%20stream%2C%20it%20is%20giving%20an%20error%20due%20to%20this%20change.,-This%20is%20the)

Reply:
I think this is because the save-discard widget handler is called for stream creation page as well. Maybe we can update that to only be called for the stream edit page in a prep commit.

We can still keep this condition since it this class is only used for the save-discard handler and nothing else like CSS or something, but the class name is such that it does not seem only related to the save-discard widget someone reading the code might feel why is this class included only for stream settings.

Reply:
Or anyways might be good to just have separate files for the permissions section and advanced configuration section with heading and save discard widget code being in the `stream_settings.hbs. But I guess that would be good as a follow-up and not worth of blocking this further.

So, I think you can just keep this as it is for now and we can handle it later.